sdk/tests/core/test_az_json.c

```diff
@@ -351,27 +412,69 @@ static void test_json_parser(void** state)
     TEST_EXPECT_SUCCESS(az_json_parser_next_token(&parser));
     test_json_token_helper(
         parser.token, AZ_JSON_TOKEN_NUMBER, AZ_SPAN_FROM_STR("10000000000000000000000e17"));
+
+    double actual_d = 0;
+    TEST_EXPECT_SUCCESS(az_json_token_get_double(&parser.token, &actual_d));
```

Comment:
should we have a test for az_json_token_get_int32 with this double value?

Reply:
We do: https://github.com/Azure/azure-sdk-for-c/pull/896/files#diff-8de5cefc7031d814e36ee0e3edc17c45R982